packages/tradershub/src/features/cfd/modals/ButtonGroups/CreateAccountButton.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Provider } from '@deriv/library';
+import { Button } from '@deriv-com/ui';
+import useDxtradeAccountHandler from '../../../../hooks/useDxtradeAccountHandler';
+import useMT5AccountHandler from '../../../../hooks/useMT5AccountHandler';
+import { validPassword } from '../../../../utils/password';
+import { MarketType, PlatformDetails, TTM5FilterLandingCompany } from '../../constants';
+
+type TCreateAccountButtonProps = {
+    buttonText: string;
+    password: string;
+    selectedJurisdiction?: TTM5FilterLandingCompany;
+};
+
+const DxtradeAccountButton = ({ buttonText, password }: TCreateAccountButtonProps) => {
+    const { createDxtradeAccountLoading, handleSubmit } = useDxtradeAccountHandler();
+    const isLoading = createDxtradeAccountLoading;
+    const isDisabled = !password || isLoading || !validPassword(password);
+    return (
+        <Button
+            disabled={isDisabled}
+            isFullWidth
+            isLoading={isLoading}
+            onClick={() => handleSubmit(password)}
+            size='lg'
+        >
+            {buttonText}
+        </Button>
+    );
+};
+
+const MT5AccountButton = ({ buttonText, password }: TCreateAccountButtonProps) => {
+    const { getCFDState } = Provider.useCFDContext();
+    const marketType = getCFDState('marketType') ?? MarketType.ALL;
+    const selectedJurisdiction = getCFDState('selectedJurisdiction') as TTM5FilterLandingCompany;
+    const { createMT5AccountLoading, handleSubmit, tradingPlatformPasswordChangeLoading } = useMT5AccountHandler({
+        marketType,
+        selectedJurisdiction,
+    });
+    const isLoading = tradingPlatformPasswordChangeLoading || createMT5AccountLoading;
+    const isDisabled = !password || isLoading || !validPassword(password);
+
+    return (
+        <Button
+            disabled={isDisabled}
+            isFullWidth
+            isLoading={isLoading}
+            onClick={() => handleSubmit(password)}
+            size='lg'
+        >
+            {buttonText}
+        </Button>
+    );
+};
```

Comment:
Can this be a single reusable component as you are using same line of code for MT5AccountButton and DxtradeAccountButton

Reply:
definitely!

Reply:
the solution seems to make the code longer and unreadable:

1) have to define type for the chunk of button code

```JavaScript
type TReusedButtonProps = {
    buttonText: string;
    handleSubmit: (password: string) => void;
    isDisabled: boolean;
    isLoading: boolean;
    password: string;
};
```

2) define the reused button
```JavaScript
const ReusedButton = ({ buttonText, handleSubmit, isDisabled, isLoading, password }: TReusedButtonProps) => (
    <Button disabled={isDisabled} isFullWidth isLoading={isLoading} onClick={() => handleSubmit(password)} size='lg'>
        {buttonText}
    </Button>
);
```

3) and then call that reused button in each `DxtradeAccountButton` and `MT5AccountButton` like
```JavaScript
const DxtradeAccountButton = ({ buttonText, password }: TCreateAccountButtonProps) => {
// some existing code
return (
        <AccountButton
            buttonText={buttonText}
            handleSubmit={handleSubmit}
            isDisabled={isDisabled}
            isLoading={isLoading}
            password={password}
        />
    );
```

the code grew by 10+ lines which I though returning the <Button> component as it is, will suffice